src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -757,7 +757,23 @@ public override void DidReceiveChallenge (NSUrlSession session, NSUrlSessionTask
 
 				if (sessionHandler.Credentials != null && TryGetAuthenticationType (challenge.ProtectionSpace, out string authType)) {
 					NetworkCredential credentialsToUse = null;
-					if (authType != RejectProtectionSpaceAuthType) {
+					// interesting situation, when we use a credential that we created that is empty, we are not getting the RejectProtectionSpaceAuthType,
+					// nevertheless, we need to check is not the first challange we will continue trusting the 
+					// TryGetAuthenticationType method, but we will also check that the status respose in no a 401
+					// look like we do get an exception from the credentials db:
+					//  TestiOSHttpClient[28769:26371051] CredStore - performQuery - Error copying matching creds.  Error=-25300, query={
+					// class = inet;
+					// "m_Limit" = "m_LimitAll";
+					// ptcl = htps;
+					// "r_Attributes" = 1;
+					// sdmn = test;
+					// srvr = "jigsaw.w3.org";
+					// sync = syna;
+					// }
+					// do remember that we ALWAYS get a challange, so the failure count has to be 1 or more for this check, 1 would be the first time
+					var nsurlRespose = challenge.FailureResponse as NSHttpUrlResponse;
+					var responseIsUnauthorized = (nsurlRespose == null) ? false : nsurlRespose.StatusCode == (int) HttpStatusCode.Unauthorized && challenge.PreviousFailureCount > 0;
+					if (authType != RejectProtectionSpaceAuthType && !responseIsUnauthorized) {
```

Comment:
nitpick: if `authType == RejectProtectionSpaceAuthType` then you're computing all this new stuff for nothing
might be better to nest the new code under the existing `if (authType != RejectProtectionSpaceAuthType) {`

Reply:
good point